src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs

```diff
@@ -325,7 +333,8 @@ private partial class CompilationTracker : ICompilationTracker
                                 generatedSource.SyntaxTree.Options,
                                 ProjectState.LanguageServices,
                                 // Compute the checksum on demand from the given source text.
-                                originalSourceTextChecksum: null));
+                                originalSourceTextChecksum: null,
+                                generationDateTime));
```

Comment:
these are the places in the system where the generation actually happens, and so we record the time it happened as well.  From this point on, this just flows outwards to be read and stored and eventually shown in the UI.